event/queue/eventQueue.go

```diff
@@ -0,0 +1,157 @@
+package queue
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"sync"
+
+	"github.com/go-kit/kit/log"
+	"github.com/xmidt-org/glaukos/event/parsing"
+	"github.com/xmidt-org/webpa-common/basculechecks"
+	"github.com/xmidt-org/webpa-common/logging"
+	"github.com/xmidt-org/webpa-common/semaphore"
+	"github.com/xmidt-org/wrp-go/v3"
+	"go.uber.org/fx"
+)
+
+var (
+	defaultLogger = log.NewNopLogger()
+)
+
+const (
+	defaultMinMaxWorkers = 5
+	defaultMinQueueSize  = 5
+)
+
+// QueueConfig configures the glaukos queue used to parse incoming events from Caduceus
+type Config struct {
+	QueueSize  int
+	MaxWorkers int
+}
+
+type EventQueue struct {
+	queue   chan wrp.Message
+	workers semaphore.Interface
+	wg      sync.WaitGroup
+	logger  log.Logger
+	config  Config
+	parsers parsing.ParsersIn
+	metrics QueueMetricsIn
+}
+
+func newEventQueue(config Config, parsers parsing.ParsersIn, metrics QueueMetricsIn, logger log.Logger) (*EventQueue, error) {
+	if parsers.BootTimeParser == nil {
+		return nil, errors.New("No boot time parser")
+	}
+
+	if parsers.MetadataParser == nil {
+		return nil, errors.New("No metadata parser")
+	}
+
+	if config.MaxWorkers < defaultMinMaxWorkers {
+		config.MaxWorkers = defaultMinMaxWorkers
+	}
+
+	if config.QueueSize < defaultMinQueueSize {
+		config.QueueSize = defaultMinQueueSize
+	}
+
+	if logger == nil {
+		logger = defaultLogger
+	}
+
+	queue := make(chan wrp.Message, config.QueueSize)
+	workers := semaphore.New(config.MaxWorkers)
+
+	e := EventQueue{
+		config:  config,
+		queue:   queue,
+		logger:  logger,
+		workers: workers,
+		parsers: parsers,
+		metrics: metrics,
+	}
+
+	return &e, nil
+}
+
+// ProvideEventQueue creates a new queue and appends the start and stop functions to the uber/fx lifecycle.
+func ProvideEventQueue(config Config, lc fx.Lifecycle, parsers parsing.ParsersIn, metrics QueueMetricsIn, logger log.Logger) (*EventQueue, error) {
```

Comment:
`Provide` is meant to be a keyword for specific `uber fx` functions that return `fx.Option`, so I think we should name this something else.  I'm not sure if we currently have a convention for this type of function - @joe94 or @kcajmagic might know.